aws-api-appsync/src/main/java/com/amplifyframework/datastore/appsync/ModelConverter.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.datastore.appsync;
+
+import com.amplifyframework.core.model.Model;
+import com.amplifyframework.util.GsonFactory;
+import com.amplifyframework.util.GsonObjectConverter;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Utility for converting a Model to/from a Map&lt;String, Object&gt;.
+ */
+public final class ModelConverter {
+
+    private ModelConverter() { }
```

Comment:
nit: we don't enforce "whitespace around" checkstyle rules for empty constructors so we can just do `{}` instead of `{ }`